src/node_file.cc

```diff
@@ -479,9 +479,9 @@ void FillStatsArray(double* fields, const uv_stat_t* s) {
   fields[9] = -1;
 #endif
   // Dates.
-#define X(idx, name)                          \
-  fields[idx] = (s->st_##name.tv_sec * 1e3) + \
-                (s->st_##name.tv_nsec / 1e6); \
+#define X(idx, name)                                                 \
+  fields[idx] = (static_cast<uint32_t>(s->st_##name.tv_sec) * 1e3) + \
+                (static_cast<uint32_t>(s->st_##name.tv_nsec) / 1e6); \
```

Comment:
> This whole fix turns Y2K38 to something like Y3K16

Ah, right. (Still not great, but yes, it’s better.)

> But there's a bug when doing `static_cast<uint64_t>` since the MSB is 1, it blows up to a huge number.

You have the overflow problem anyway, though? Also, this doesn’t happen if you cast to `double` directly.